docs/xml.md

```diff
@@ -1,6 +1,681 @@
+
+The following attributes are REQUIRED for all property elements, regardless of their type or name:
+
+
+| Name   | Contents                                              |
+|:-------|:------------------------------------------------------|
+| `name` | The name of the property represented by this element. |
+
+The `name` of a property does not necessarily reflect its in-engine name, and some properties serialize with unexpected names. Additionally, not every property is deserialized by Roblox.
+
+The format for each data type is listed below.
+
+### Axes
+
+The `Axes` data type is represented with a single `axes` element that contains a single integer between `0` and `7`. This integer represents a bitfield of the `Z`, `Y`, and `X` axes packed into the lower 3 bits of it, in that order.
+
+An `Axes` property with only the `X` axis enabled appears as follows:
+
+```xml
+<Axes name="AxesExample">
+	<axes>1</axes>
+</Axes>
+```
+
+### BinaryString
+
+The `BinaryString` data type is represented by the contents of the property encoded with Base64.
+
+A `BinaryString` property with the contents `Rojo is cool!` appears as follows:
+
+```xml
+<BinaryString name="BinaryStringExample">Um9qbyBpcyBjb29sIQ==</BinaryString>
+```
+
+### bool
+
+The `bool` data type is represented by a literal string reading either `true` or `false` depending upon the state of the value.
+
+Although Roblox accepts variations such as `fAlSe` and `TRUE`, by convention values SHOULD be written in all lowercase.
+
+A `bool` with the value `false` appears as follows:
+
+```xml
+<bool name="BoolExample">false</bool>
+```
+
+### BrickColor
+
+The `BrickColor` data type is represented by a single 32-bit integer that represents the `Number` of the value.
+
+Roblox encodes this type with the element name `int` but also accepts `BrickColor`. It is preferred that `BrickColor` be used.
+
+A `BrickColor` with the value `Medium Stone Grey` (whose number is `194`) should appear as follows:
+
+```xml
+<BrickColor name="BrickColorExample">194</BrickColor>
+```
+
+### Color3
+
+The `Color3` data type is represented by three child elements named `R`, `G`, and `B`. These elements contain the value of that component as written as a [`float`](#float).
+
+A `Color3` with the value `INF, 1337, 0.15625` appears as follows:
+
+```xml
+<Color3 name="Color3Example">
+	<R>INF</R>
+	<G>1337</G>
+	<B>0.15625</B>
+</Color3>
+```
+
+### Color3uint8
+
+The `Color3uint8` data type is represented by a single unsigned 32-bit integer that is the `R`, `G`, and `B` components of the color (as integers in the range 0 to 255) packed into the lower 24 bits of the number. This integer is little-endian and written in the order `G`, `B`, `R`.
+
+The upper 8 bits of the value SHOULD be filled with `FF` (in hexadecimal). This is to maintain compatibility with Roblox.
+
+A `Color3uint8` with the value `96, 64, 32` appears as follows:
```

Comment:
Might be nice to show the steps in getting to `4284497952`, like the above

Reply:
I've done my best to specify but I'm not sure how best to explain it while keeping tonally consistent. On the plus side, I discovered an issue with the info I'd written while explaining, so that's fixed now.